libs/core/src/config.ts

```diff
@@ -32,33 +50,86 @@ const {
   TEST_WITHOUT_LOGS,
 } = process.env;
 
-const PORT = _PORT ? parseInt(_PORT, 10) : 8080;
+const DEFAULTS = {
+  NODE_ENV: 'development',
+  PORT: 8080,
+  get SERVER_URL() {
+    return `http://localhost:${this.PORT}`;
+  },
+  ROLLBAR_ENV: 'local',
+  ROLLBAR_SERVER_TOKEN: '',
+};
 
 export const config = configure(
   {},
   {
-    NODE_ENV: (NODE_ENV || 'development') as Environment,
+    APP_ENV: APP_ENV as AppEnvironment,
+    APP_ENV_PASSWORD: APP_ENV_PASSWORD,
+    NODE_ENV: (NODE_ENV || DEFAULTS.NODE_ENV) as Environment,
     IS_CI: IS_CI === 'true',
-    SERVER_URL:
-      SERVER_URL ?? NODE_ENV === 'production'
-        ? 'https://commonwealth.im'
-        : `http://localhost:${PORT}`,
-    PORT,
+    SERVER_URL: SERVER_URL ?? DEFAULTS.SERVER_URL,
+    PORT: _PORT ? parseInt(_PORT, 10) : DEFAULTS.PORT,
     LOGGING: {
-      ROLLBAR_SERVER_TOKEN: _ROLLBAR_SERVER_TOKEN || '',
-      ROLLBAR_ENV: _ROLLBAR_ENV || 'local',
+      ROLLBAR_SERVER_TOKEN:
+        _ROLLBAR_SERVER_TOKEN || DEFAULTS.ROLLBAR_SERVER_TOKEN,
+      ROLLBAR_ENV: _ROLLBAR_ENV || DEFAULTS.ROLLBAR_ENV,
       TEST_WITHOUT_LOGS: TEST_WITHOUT_LOGS === 'true',
     },
   },
   z.object({
+    APP_ENV: z.enum(AppEnvironments).refine(() => {
+      // prevent APP_ENV from being set to 'production' without the correct password
```

Comment:
What's the intent for requiring a password in this case?

Reply:
To prevent as much as possible setting `APP_ENV=production` in non-production environments. `APP_ENV` is meant to be strict in the sense that `APP_ENV=production` in a local environment would produce dangerous results e.g. emit notifications/emails to real users. This differs from `NODE_ENV=production` which should mainly be used to modify logs, build processes, etc.

Reply:
gotcha